### PR TITLE
WIP Authorization

### PR DESCRIPTION
**Pull request by Yar3k**

### CODE REVIEW

File: Support_Your_Locals/Controllers/AuthController.cs [hunk 1]
```diff
@@ -64,7 +69,7 @@ public ViewResult SignIn()
         }
 
         [HttpPost]
-        public ActionResult SignIn(UserLoginModel login)
+        public async Task<ActionResult> SignIn(UserLoginModel login)
```
Comment (EvalVis) [inline Support_Your_Locals/Controllers/AuthController.cs:73]: Fantastic. If I understand correctly, we now fulfill a requirement to use concurrent programming

File: Support_Your_Locals/Controllers/AuthController.cs [hunk 2]
```diff
@@ -97,12 +98,11 @@ public async Task<ActionResult> SignIn(UserLoginModel login)
                         {
```
Comment (EvalVis) [inline Support_Your_Locals/Controllers/AuthController.cs:98]: One question: why claims are list? Should not it be one username who is logged in? Or do I not understand something correctly?

**Review by EvalVis — APPROVED** [review]
Good job. Just some insights:
1) Tip for the future: You registered to this branch from add-timesheets (or how it is called?..) However, main branch should be used instead because now there are unnecessary code duplicates (and that can probably cause problems) like Breakthrough, Remake timepicker (which are in that branch also).
2) You should really consider creating a separate authentication class(es). This code in one action is huge.

P.S. Did I miss, or there is no redirection when you enter /SignIn or /SingUp in browser search bar.
If so, there should be redirection to home page or account settings, for example.

File: Support_Your_Locals/Controllers/AuthController.cs (hunk 2 above)
Comment (Yar3k) [reply Support_Your_Locals/Controllers/AuthController.cs:98]: Well everyone uses claims list sooo .... 🤣🤣🤣

**Pull request merged**